profiles/common/modules/custom/nexteuropa_varnish/src/PurgeRuleType.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * @file
+ * Definition of Drupal\nexteuropa_varnish\PurgeRuleType.
+ */
+
+namespace Drupal\nexteuropa_varnish;
+
+/**
+ * Enum of purge rule types.
+ */
+class PurgeRuleType {
+  const NODE = 'node';
+  const PATHS = 'paths';
+
+  /**
+   * Prevents the class from instantiated, it just acts as an enum.
+   */
+  private function __construct() {
```

Comment:
It's not a singleton. It's a basic lightweight implementation of an enumeration, containing valid purge rule types:

PurgeRuleType::NODE
PurgeRuleType::PATHS